File: chatosdk/src/main/java/com/gamatechno/chato/sdk/app/chatrooms/ChatRoomsView.java
package com.gamatechno.chato.sdk.app.chatrooms;

import com.gamatechno.chato.sdk.app.chatrooms.uimodel.ChatRoomsUiModel;
import com.gamatechno.chato.sdk.module.core.BaseView;

import java.util.List;

public interface ChatRoomsView {
    interface ObrolanPresenter{
        void requestObrolan(boolean isRefresh, String keyword);
        void pinChatRoom(ChatRoomsUiModel chatRoomUiModel, int total);
        void deleteRoom(ChatRoomsUiModel chatRoomUiModel);
    }

    interface View extends BaseView{
        void onRequestObrolan(List<ChatRoomsUiModel> list, boolean isRefresh);
        void onFailedRequestObrolan();
        void successPinnedChatRoom(String message);
        void onDeleteRoom(boolean isSuccess, String message);
    }
 }

File: chatosdk/src/main/java/com/gamatechno/chato/sdk/app/chatrooms/ChatRoomsView.kt
package com.gamatechno.chato.sdk.app.chatrooms

import com.gamatechno.chato.sdk.app.chatrooms.uimodel.ChatRoomsUiModel
import com.gamatechno.chato.sdk.module.core.BaseView

interface ChatRoomsView {
    interface ObrolanPresenter {
        fun requestObrolan(isRefresh: Boolean, keyword: String?, categories: String)
        fun pinChatRoom(chatRoomUiModel: ChatRoomsUiModel?, total: Int)
        fun deleteRoom(chatRoomUiModel: ChatRoomsUiModel?)
    }

    interface View : BaseView {
        fun onRequestObrolan(list: List<ChatRoomsUiModel?>?, isRefresh: Boolean)
        fun onFailedRequestObrolan()
        fun successPinnedChatRoom(message: String?)
        fun onDeleteRoom(isSuccess: Boolean, message: String?)
    }
}